tools/rebuild-elasticsearch-transfer-index-from-files

```diff
@@ -18,148 +18,149 @@
-
-    return found
 
-conn = pyes.ES(elasticSearchFunctions.getElasticsearchServerHostAndPort())
+def main():
+    # Make sure the user knows what he or she is going to do
+    print('WARNING: This script will delete your current ElasticSearch transfer data, rebuilding it using files.')
+    proceed = raw_input('Are you sure you want to continue? (yes/no)\n')
+    if proceed != 'yes':
+        print("You didn't enter 'yes': exiting.")
+        sys.exit(0)
 
-try:
-    conn._send_request('GET', '')
-except pyes.exceptions.NoServerAvailable:
-    print "Connection error: Is Elasticsearch running?"
-    os._exit(1)
+    # Determine root of shared directories
+    clientConfigFilePath = '/etc/archivematica/MCPClient/clientConfig.conf'
+    config = ConfigParser.SafeConfigParser()
+    config.read(clientConfigFilePath)
 
-# delete transfers ElasticSearch index
-try:
-    conn.delete_index('transfers')
-except pyes.exceptions.IndexMissingException:
-    pass
+    try:
+        sharedDirectory = config.get('MCPClient', 'sharedDirectoryMounted')
+    except:
+        print('Configuration item "sharedDirectoryMounted" not available at /etc/archivematica/MCPClient/clientConfig.conf.')
+        sys.exit(1)
 
-# recreate transfers index, setting up mapping
-elasticSearchFunctions.check_server_status_and_create_indexes_if_needed()
+    # Set root directory
+    try:
+        rootdir = sys.argv[1]
+        if not os.path.exists(rootdir):
+            print("AIP store location doesn't exist.")
+            sys.exit(1)
+    except IndexError:
+        print('Usage: am rebuild-elasticsearch-transfer-index-from-files <path to AIP store>')
+        rootdir = os.path.join(sharedDirectory, 'www/AIPsStore')
+        print('Default path is: ' + rootdir)
+        sys.exit(1)
+
+    print('Rebuilding transfer index from transfer backlog in', rootdir)
+
+    # Verify ES is accessible
+    try:
+        elasticSearchFunctions.setup_reading_from_client_conf()
+        es_client = elasticSearchFunctions.get_client()
+    except AttributeError:
+        es_client = elasticSearchFunctions.Elasticsearch(hosts=elasticSearchFunctions.getElasticsearchServerHostAndPort())  # AM 1.5
 
-temp_dir = tempfile.mkdtemp()
+    try:
+        es_client.info()
+    except Exception:
+        print("Error: Elasticsearch may not be running.")
+        sys.exit(1)
+
+    print('Deleting all transfers in the "transfers" index')
+    time.sleep(3)  # Time for the user to panic and kill the process
+    es_client.indices.delete('transfers', ignore=404)
+    try:
+        elasticSearchFunctions.create_indexes_if_needed(es_client)
+    except AttributeError:
+        elasticSearchFunctions.check_server_status_and_create_indexes_if_needed()  # AM 1.5
 
-backlog_original_dir = os.path.join(rootdir, 'transferBacklog', 'originals')
+    temp_dir = tempfile.mkdtemp()
+    backlog_original_dir = os.path.join(rootdir, 'transferBacklog', 'originals')
```

Comment:
I don't think you can assume anymore that the Transfer Backlog is a subdirectory of AIP Storage.  The Storage Service allows you to configure Transfer Backlog completely separately.  So I think you have to just use the path the user provides, and ask them to provide the full path to the transfer backlog.